mindsdb/integrations/handlers/ray_serve_handler/ray_serve_handler.py

```diff
@@ -29,10 +29,23 @@ def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[
         # TODO: use join_learn_process to notify users when ray has finished the training process
         args = args['using']  # ignore the rest of the problem definition
         args['target'] = target
+
+        # Store the args in model storage for later use
         self.model_storage.json_set('args', args)
+
+        # Prepare the payload for training, including additional parameters
+        payload = {
+            'df': df.to_json(orient='records'),
+            'target': target
+        }
+
+        # Include additional parameters if they exist
+        if 'learn' in args:
+            payload['learn'] = args['learn']
```

Comment:
User's query  to create model will be:
```sql
 CREATE MODEL rayserve_model
    FROM example_db (SELECT * FROM home_rentals)
    PREDICT location
    USING
    engine='ray_serve',
    train_url='http://127.0.0.1:8000/my_model/train',
    predict_url='http://127.0.0.1:8000/my_model/predict',
    custom_arg1=1,
    custom_arg1='asdf';
```
All arguments are located in args parameter. What to do: cut standard arguments (train_url, predict_url) and send the rest as args parameter to ray server

Reply:
sample code:
```python
args = copy.copy(args)
args.pop('train_url')
args.pop('predict_url')

resp = requests.post(args['train_url'],  json={'df': df.to_json(orient='records'), 'target': target, 'args': args}, ....
```

Reply:
```
   import copy
    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
    # Copy the `args` object to avoid modifying the original
    args = copy.copy(args['using'])

    
    # Extract standard arguments
    train_url = args.pop('train_url', None)
    predict_url = args.pop('predict_url', None)
    
    if not train_url or not predict_url:
        raise Exception("Error: Both 'train_url' and 'predict_url' must be provided.")

    # Now `args` contains only custom arguments (e.g., `custom_arg1`, `custom_arg2`)
    args['target'] = target

    self.model_storage.json_set('args', {'train_url': train_url, 'predict_url': predict_url, 'custom_args': args})

    # Prepare the request payload
    payload = {
        'df': df.to_json(orient='records'),
        'target': target,
        'args': args  # Pass custom arguments here
    }
    
    try:
        # Send the training request to the Ray Serve API
        resp = requests.post(
            train_url,
            json=payload,
            headers={'content-type': 'application/json; format=pandas-records'}
        )
        resp.raise_for_status()  # Raise an exception if the request fails
    except requests.exceptions.RequestException as e:
        raise Exception(f"Error: Failed to communicate with the training endpoint. Details: {str(e)}")

    # Handle the API response
    response = resp.json()
    if response.get('status') != 'ok':
        raise Exception(f"Error: Training failed. Details: {response.get('status', 'Unknown error')}")
```



Hey @ea-rus kindly look into this code, will this work, I have separated the standard arguments from the custom ones, and send the custom arguments to the Ray serve under `"args"`  key

Reply:
It should work, but I don't know do we need to rename it to custom_args, maybe to keep it as it is (store original arguments)
`self.model_storage.json_set('args', {'train_url': train_url, 'predict_url': predict_url, 'custom_args': args})`

Reply:
ok... then I'll make these changes and then create a new PR